tests/ProxyManagerTest/ProxyGenerator/AccessInterceptorScopeLocalizer/MethodGenerator/BindProxyPropertiesTest.php

```diff
@@ -198,116 +154,53 @@ public function testBodyStructureWithTypedProperties(): void
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateBoolProperty = & $localizedObject->privateBoolProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableBoolProperty = & $localizedObject->privateNullableBoolProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateIntProperty = & $localizedObject->privateIntProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableIntProperty = & $localizedObject->privateNullableIntProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateFloatProperty = & $localizedObject->privateFloatProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableFloatProperty = & $localizedObject->privateNullableFloatProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateStringProperty = & $localizedObject->privateStringProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableStringProperty = & $localizedObject->privateNullableStringProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateArrayProperty = & $localizedObject->privateArrayProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableArrayProperty = & $localizedObject->privateNullableArrayProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateIterableProperty = & $localizedObject->privateIterableProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
-\Closure::bind(function () use ($localizedObject) {
-    $this->privateNullableIterableProperty = & $localizedObject->privateNullableIterableProperty;
-}, $this, 'ProxyManagerTestAsset\\ClassWithMixedReferenceableTypedProperties')->__invoke();
-
+        $expectedCode = <<<'PHP'
+$class = new \ReflectionObject($localizedObject);
```

Comment:
Ideally, we should be able to avoid runtime reflection for all properties that do not have a declared type or do have a default `null` value

Reply:
Done - I returned original version of code generation for untyped properties and nullable properties with default values.

Other properties (`$nonReferenceableProperties`) are then used to generate localization code with `ReflectionProperty::isInitialized()` check.